source/code/plugins/in_npmd_server.rb

```diff
@@ -353,6 +361,30 @@ def is_valid_dataitem(item)
             end
         end
 
+        def shouldUploadNetworkAgentInfo(item)
+            _retVal = false
+            if(@networkAgentState == nil || @networkAgentState["SubnetId"] == nil || @networkAgentState["AgentIP"] == nil || @networkAgentState["AgentFqdn"] == nil || @networkAgentState["TimeGenerated"] == nil) 
+                _retVal = true
+            end
+            _timeDiff = 0
+            if(@networkAgentState != nil && @networkAgentState["TimeGenerated"] != nil && item["TimeGenerated"] != nil)
+                _timeDiff = Time.parse(item["TimeGenerated"]) - Time.parse(@networkAgentState["TimeGenerated"])
+                _timeDiff /= 3600
+            end
+            if(!_retVal && _timeDiff > 6 || _timeDiff < 0)
+                _retVal = true
+            end
+            if(!_retVal && (@networkAgentState["SubnetId"] != item["SubnetId"] || @networkAgentState["AgentIP"] != item["AgentIP"] || @networkAgentState["AgentFqdn"] != item["AgentFqdn"]))
+                _retVal = true
+            end
+            if(_retVal)
+                @netWorkAgentSync.synchronize do
```

Comment:
Ideally if we need to protect the networkAgentState variable then have this synchronize for the logic where we are reading as well as updating it.